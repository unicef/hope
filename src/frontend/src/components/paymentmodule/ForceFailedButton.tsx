import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogContainer } from '@containers/dialogs/DialogContainer';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { useMarkPayAsFailedMutation } from '@generated/graphql';

export interface ForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function ForceFailedButton({
  paymentId,
  disabled = false,
}: ForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useMarkPayAsFailedMutation();

  const submit = async (): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
        },
      });
      setOpenModal(false);
      showMessage(t('Payment has been marked as failed.'));
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-mark-as-failed"
          disabled={disabled}
        >
          {t('Mark as failed')}
        </Button>
      </Box>
      <Dialog
        open={isOpenModal}
        onClose={() => setOpenModal(false)}
        scroll="paper"
        aria-labelledby="form-dialog-title"
        maxWidth="md"
      >
        <DialogTitleWrapper>
          <DialogTitle>{t('Mark as failed')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t('Are you sure you would like to mark payment as failed?')}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setOpenModal(false)}>{t('CANCEL')}</Button>
            <Button
              type="submit"
              color="primary"
              variant="contained"
              onClick={() => submit()}
              data-cy="button-submit"
              disabled={loading}
            >
              {t('Mark as failed')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </Box>
  );
}
