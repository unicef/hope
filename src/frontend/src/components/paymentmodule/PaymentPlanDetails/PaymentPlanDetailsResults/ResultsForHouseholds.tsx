import { Grid2 as Grid } from '@mui/material';
import {
  SummaryBorder,
  SummaryValue,
} from '@components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsResults/Styles';
import { LabelizedField } from '@core/LabelizedField';
import { PaymentPlanQuery } from '@generated/graphql';
import { useProgramContext } from 'src/programContext';

interface ResultsForHouseholdsProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const ResultsForHouseholds = ({
  paymentPlan,
}: ResultsForHouseholdsProps) => {
  const { totalHouseholdsCount, totalIndividualsCount } = paymentPlan;
  const { selectedProgram } = useProgramContext();
  const beneficiaryGroup = selectedProgram?.beneficiaryGroup;

  return (
    <Grid size={{ xs: 4 }}>
      <Grid container spacing={0} justifyContent="flex-end">
        <Grid size={{ xs: 6 }}>
          <SummaryBorder>
            <LabelizedField
              label={`Total Number of ${beneficiaryGroup?.groupLabelPlural}`}
            >
              <SummaryValue>{totalHouseholdsCount || '0'}</SummaryValue>
            </LabelizedField>
          </SummaryBorder>
        </Grid>
        <Grid size={{ xs: 6 }}>
          <SummaryBorder>
            <LabelizedField
              label={`Targeted ${beneficiaryGroup?.memberLabelPlural}`}
            >
              <SummaryValue>{totalIndividualsCount || '0'}</SummaryValue>
            </LabelizedField>
          </SummaryBorder>
        </Grid>
      </Grid>
    </Grid>
  );
};
