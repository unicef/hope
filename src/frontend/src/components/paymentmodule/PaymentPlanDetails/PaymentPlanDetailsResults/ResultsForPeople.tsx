import { PaymentPlanQuery } from '@generated/graphql';
import { useTranslation } from 'react-i18next';
import { Grid } from '@mui/material';
import {
  SummaryBorder,
  SummaryValue,
} from '@components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsResults/Styles';
import { LabelizedField } from '@core/LabelizedField';
import * as React from 'react';

interface ResultsForPeopleProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const ResultsForPeople = ({ paymentPlan }: ResultsForPeopleProps) => {
  const { t } = useTranslation();
  const { totalHouseholdsCount } = paymentPlan;

  return (
    <Grid item xs={4}>
      <Grid container spacing={0} justifyContent="flex-end">
        <Grid item>
          <SummaryBorder>
            <LabelizedField label={t('Total Number of People')}>
              <SummaryValue>{totalHouseholdsCount || '0'}</SummaryValue>
            </LabelizedField>
          </SummaryBorder>
        </Grid>
      </Grid>
    </Grid>
  );
};
