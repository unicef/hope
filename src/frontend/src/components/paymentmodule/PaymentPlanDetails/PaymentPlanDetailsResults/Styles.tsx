import styled from 'styled-components';
import { MiśTheme } from '../../../../theme';

export const colors = {
  femaleChildren: '#5F02CF',
  maleChildren: '#1D6A64',
  femaleAdult: '#DFCCF5',
  maleAdult: '#B1E3E0',
};
export const Title = styled.div`
  padding-bottom: ${({ theme }) => theme.spacing(2)};
`;
export const ContentWrapper = styled.div`
  display: flex;
`;
export const SummaryBorder = styled.div`
  padding: ${({ theme }) => theme.spacing(4)};
  border-color: #b1b1b5;
  border-left-width: 1px;
  border-left-style: solid;
`;
export const SummaryValue = styled.div`
  font-family: ${({ theme }: { theme: MiśTheme }) =>
    theme.hctTypography.fontFamily};
  color: #253b46;
  font-size: 36px;
  line-height: 32px;
  margin-top: ${({ theme }) => theme.spacing(2)};
`;
export const ChartContainer = styled.div`
  width: 100px;
  height: 100px;
  margin: 0 auto;
`;
