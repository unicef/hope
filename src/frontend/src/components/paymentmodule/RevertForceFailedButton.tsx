import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Field, Form, Formik } from 'formik';
import * as Yup from 'yup';
import CalendarTodayRoundedIcon from '@mui/icons-material/CalendarTodayRounded';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { FormikTextField } from '@shared/Formik/FormikTextField';
import { FormikDateField } from '@shared/Formik/FormikDateField';
import { useRevertMarkPayAsFailedMutation } from '@generated/graphql';

export interface RevertForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function RevertForceFailedButton({
  paymentId,
  disabled = false,
}: RevertForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useRevertMarkPayAsFailedMutation();

  const validationSchema = Yup.object().shape({
    deliveredQuantity: Yup.number()
      .min(0)
      .max(99999999, t('Number is too big')),
    deliveryDate: Yup.date().required(t('Delivery date is required')),
  });

  const submit = async (values, { resetForm }): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
          deliveredQuantity: values.deliveredQuantity,
          deliveryDate: values.deliveryDate,
        },
      });
      setOpenModal(false);
      showMessage(t('Force failed has been reverted.'));
      resetForm();
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-revert-mark-as-failed"
          disabled={disabled}
        >
          {t('Revert mark as failed')}
        </Button>
      </Box>
      <Formik
        initialValues={{
          deliveredQuantity: 0,
          deliveryDate: '',
        }}
        validationSchema={validationSchema}
        onSubmit={submit}
      >
        {({ submitForm, resetForm }) => (
          <Dialog
            open={isOpenModal}
            onClose={() => setOpenModal(false)}
            scroll="paper"
            aria-labelledby="form-dialog-title"
            maxWidth="md"
          >
            <DialogTitleWrapper>
              <DialogTitle>{t('Revert mark as failed')}</DialogTitle>
            </DialogTitleWrapper>
            <DialogContent>
              <Form>
                <Field
                  name="deliveredQuantity"
                  label={t('Delivered Quantity')}
                  type="number"
                  required
                  fullWidth
                  variant="outlined"
                  component={FormikTextField}
                />
                <Field
                  name="deliveryDate"
                  label={t('Delivery Date')}
                  component={FormikDateField}
                  required
                  fullWidth
                  decoratorEnd={<CalendarTodayRoundedIcon color="disabled" />}
                />
              </Form>
            </DialogContent>
            <DialogFooter>
              <DialogActions>
                <Button
                  onClick={() => {
                    setOpenModal(false);
                    resetForm();
                  }}
                >
                  {t('CANCEL')}
                </Button>
                <Button
                  type="submit"
                  color="primary"
                  variant="contained"
                  onClick={submitForm}
                  data-cy="button-submit"
                  disabled={loading}
                >
                  {t('Revert mark as failed')}
                </Button>
              </DialogActions>
            </DialogFooter>
          </Dialog>
        )}
      </Formik>
    </Box>
  );
}
