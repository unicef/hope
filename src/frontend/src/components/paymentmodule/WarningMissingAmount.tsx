import { Box } from '@mui/material';
import WarningIcon from '@mui/icons-material/Warning';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { ReactElement } from 'react';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): ReactElement {
  const { t } = useTranslation();

  return (
    <Box display="flex" alignItems="center">
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
