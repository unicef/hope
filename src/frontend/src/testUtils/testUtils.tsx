import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import { Formik } from 'formik';
import noop from 'lodash/noop';
import { TestProviders } from './testProviders';
import { ProgramStatus } from '@generated/graphql';
import { MockLink } from '@apollo/client/testing';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(
    <>
      <BrowserRouter>
        <Formik initialValues={{}} onSubmit={noop}>
          {ui}
        </Formik>
      </BrowserRouter>
    </>,
    {
      wrapper: TestProviders,
      ...options,
    },
  );

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export class ApolloLoadingLink extends MockLink {
  constructor() {
    super([]);
  }

  request() {
    return null;
  }
}

export const fakeContextProgram = {
  selectedProgram: {
    id: '1',
    name: 'someName',
    status: ProgramStatus.Active,
    pduFields: null,
    dataCollectingType: {
      id: '1',
      householdFiltersAvailable: true,
      individualFiltersAvailable: true,
      label: 'data collecting type',
      code: '123',
      type: 'full',
      children: null,
    },
  },
  setSelectedProgram: () => {},
  isActiveProgram: true,
  isSocialDctType: false,
  isStandardDctType: true,
  programHasPdu: false,
};
