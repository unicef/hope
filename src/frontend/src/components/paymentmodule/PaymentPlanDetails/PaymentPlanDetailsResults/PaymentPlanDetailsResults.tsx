import { Grid, Typography } from '@mui/material';
import * as React from 'react';
import { Pie } from 'react-chartjs-2';
import { useTranslation } from 'react-i18next';
import { PaymentPlanQuery } from '@generated/graphql';
import { LabelizedField } from '@core/LabelizedField';
import { PaperContainer } from '../../../targeting/PaperContainer';
import { FieldBorder } from '@core/FieldBorder';
import { useProgramContext } from '../../../../programContext';
import {
  ChartContainer,
  colors,
  ContentWrapper,
  Title,
} from '@components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsResults/Styles';
import { ResultsForHouseholds } from '@components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsResults/ResultsForHouseholds';
import { ResultsForPeople } from '@components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsResults/ResultsForPeople';

interface PaymentPlanDetailsResultsProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const PaymentPlanDetailsResults = ({
  paymentPlan,
}: PaymentPlanDetailsResultsProps): React.ReactElement => {
  const { t } = useTranslation();
  const { isSocialDctType } = useProgramContext();
  const {
    femaleChildrenCount,
    maleChildrenCount,
    femaleAdultsCount,
    maleAdultsCount,
  } = paymentPlan;

  const ResultsComponent = isSocialDctType
    ? ResultsForPeople
    : ResultsForHouseholds;

  return (
    <PaperContainer>
      <Title>
        <Typography variant="h6">{t('Results')}</Typography>
      </Title>
      <ContentWrapper>
        <Grid container>
          <Grid item xs={4}>
            <Grid container spacing={3} justifyContent="flex-start">
              <Grid item xs={6}>
                <FieldBorder color={colors.femaleChildren}>
                  <LabelizedField
                    label={t('Female Children')}
                    value={femaleChildrenCount}
                  />
                </FieldBorder>
              </Grid>
              <Grid item xs={6}>
                <FieldBorder color={colors.femaleAdult}>
                  <LabelizedField
                    label={t('Female Adults')}
                    value={femaleAdultsCount}
                  />
                </FieldBorder>
              </Grid>
              <Grid item xs={6}>
                <FieldBorder color={colors.maleChildren}>
                  <LabelizedField
                    label={t('Male Children')}
                    value={maleChildrenCount}
                  />
                </FieldBorder>
              </Grid>
              <Grid item xs={6}>
                <FieldBorder color={colors.maleAdult}>
                  <LabelizedField
                    label={t('Male Adults')}
                    value={maleAdultsCount}
                  />
                </FieldBorder>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={4}>
            <Grid
              container
              spacing={0}
              justifyContent="flex-start"
              alignItems="center"
            >
              <Grid item xs={4}>
                <ChartContainer data-cy="chart-container">
                  <Pie
                    width={100}
                    height={100}
                    options={{
                      plugins: {
                        legend: {
                          display: false,
                        },
                      },
                      responsive: true,
                      maintainAspectRatio: false,
                    }}
                    data={{
                      labels: [
                        t('Female Children'),
                        t('Female Adults'),
                        t('Male Children'),
                        t('Male Adults'),
                      ],
                      datasets: [
                        {
                          data: [
                            femaleChildrenCount,
                            femaleAdultsCount,
                            maleChildrenCount,
                            maleAdultsCount,
                          ],
                          backgroundColor: [
                            colors.femaleChildren,
                            colors.femaleAdult,
                            colors.maleChildren,
                            colors.maleAdult,
                          ],
                        },
                      ],
                    }}
                  />
                </ChartContainer>
              </Grid>
            </Grid>
          </Grid>
          <ResultsComponent paymentPlan={paymentPlan} />
        </Grid>
      </ContentWrapper>
    </PaperContainer>
  );
};
