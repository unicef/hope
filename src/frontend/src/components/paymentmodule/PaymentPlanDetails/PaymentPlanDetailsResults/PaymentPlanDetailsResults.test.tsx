import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { fakeActionPpMutation } from '../../../../../fixtures/paymentmodule/fakeApolloActionPaymentPlanMutation';
import { render } from '../../../../testUtils/testUtils';
import { PaymentPlanDetailsResults } from './PaymentPlanDetailsResults';

describe('components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/PaymentPlanDetailsResults', () => {
  it('should render', async () => {
    const { container } = render(
      <MockedProvider addTypename={false} mocks={fakeActionPpMutation}>
        <PaymentPlanDetailsResults paymentPlan={fakeApolloPaymentPlan} />
      </MockedProvider>,
    );

    await act(() => wait(0)); // wait for the mutation to complete

    expect(container).toMatchSnapshot();
  });
});
