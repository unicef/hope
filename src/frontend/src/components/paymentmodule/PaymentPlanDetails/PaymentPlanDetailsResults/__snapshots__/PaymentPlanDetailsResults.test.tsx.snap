// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/PaymentPlanDetailsResults should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-dVBluf iGTkRx css-1gtchvp-MuiPaper-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="sc-dycHyt etyvHV"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-66ugg1-MuiTypography-root"
      >
        Results
      </h6>
    </div>
    <div
      class="sc-jbAkgO hmtbDj"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1jxwjqx-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-17yh78r-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-6d7jwz-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-ticvdu-MuiGrid2-root"
            >
              <div
                class="sc-fLDLck eqCuFP"
                color="#5F02CF"
              >
                <div>
                  <span
                    class="sc-FEMpB ddgbkI"
                    color="textSecondary"
                  >
                    Female Children
                  </span>
                  <div
                    data-cy="label-Female Children"
                  >
                    <span
                      class="sc-gjcoXW fZsJNd"
                      color="textSecondary"
                    >
                      0
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-ticvdu-MuiGrid2-root"
            >
              <div
                class="sc-fLDLck dBYgWW"
                color="#DFCCF5"
              >
                <div>
                  <span
                    class="sc-FEMpB ddgbkI"
                    color="textSecondary"
                  >
                    Female Adults
                  </span>
                  <div
                    data-cy="label-Female Adults"
                  >
                    <span
                      class="sc-gjcoXW fZsJNd"
                      color="textSecondary"
                    >
                      1
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-ticvdu-MuiGrid2-root"
            >
              <div
                class="sc-fLDLck jWyxsj"
                color="#1D6A64"
              >
                <div>
                  <span
                    class="sc-FEMpB ddgbkI"
                    color="textSecondary"
                  >
                    Male Children
                  </span>
                  <div
                    data-cy="label-Male Children"
                  >
                    <span
                      class="sc-gjcoXW fZsJNd"
                      color="textSecondary"
                    >
                      0
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-ticvdu-MuiGrid2-root"
            >
              <div
                class="sc-fLDLck qEmJv"
                color="#B1E3E0"
              >
                <div>
                  <span
                    class="sc-FEMpB ddgbkI"
                    color="textSecondary"
                  >
                    Male Adults
                  </span>
                  <div
                    data-cy="label-Male Adults"
                  >
                    <span
                      class="sc-gjcoXW fZsJNd"
                      color="textSecondary"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-17yh78r-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-xg0rmm-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-17yh78r-MuiGrid2-root"
            >
              <div
                class="sc-hRDKVd bCxRDr"
                data-cy="chart-container"
              >
                <canvas
                  height="0"
                  role="img"
                  style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                  width="0"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-17yh78r-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-fa87wz-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-ticvdu-MuiGrid2-root"
            >
              <div
                class="sc-gNZgCX hGItiC"
              >
                <div>
                  <span
                    class="sc-FEMpB ddgbkI"
                    color="textSecondary"
                  >
                    Total Number of Households
                  </span>
                  <div
                    data-cy="label-Total Number of Households"
                  >
                    <span
                      class="sc-gjcoXW fZsJNd"
                      color="textSecondary"
                    >
                      <div
                        class="sc-kfeOyU esIFmr"
                      >
                        1
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-ticvdu-MuiGrid2-root"
            >
              <div
                class="sc-gNZgCX hGItiC"
              >
                <div>
                  <span
                    class="sc-FEMpB ddgbkI"
                    color="textSecondary"
                  >
                    Targeted Individuals
                  </span>
                  <div
                    data-cy="label-Targeted Individuals"
                  >
                    <span
                      class="sc-gjcoXW fZsJNd"
                      color="textSecondary"
                    >
                      <div
                        class="sc-kfeOyU esIFmr"
                      >
                        3
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
